src/SonarQube.Client/Api/SonarQubeService.cs

```diff
@@ -123,19 +131,30 @@ public async Task ConnectAsync(ConnectionInformation connection, CancellationTok
 
             IsConnected = true;
 
+            logger.Debug($"Getting the version of SonarQube...");
+
             var versionResponse = await InvokeRequestAsync<IGetVersionRequest, string>(token);
             sonarQubeVersion = Version.Parse(versionResponse);
 
+            logger.Info($"Connected to SonarQube '{sonarQubeVersion}'.");
+
+            logger.Debug($"Validating the credentials...");
+
             var credentialResponse = await InvokeRequestAsync<IValidateCredentialsRequest, bool>(token);
             if (!credentialResponse)
             {
                 IsConnected = false;
                 throw new InvalidOperationException("Invalid credentials");
             }
+
+            logger.Debug($"Credentials accepted.");
         }
 
         public void Disconnect()
         {
+            logger.Info($"Disconnecting...");
+            logger.Debug($"IsConnected is {IsConnected}.");
```

Comment:
This message is potentially confusing: it would be clearer if it said something like 
_Current state before disconnecting is {IsConnected}._